clients/unshaded/src/main/java/tachyon/client/block/BlockStoreContext.java

```diff
@@ -55,17 +66,17 @@
    * Initializes {@link #mLocalBlockWorkerClientPool}. This method is supposed be called in a lazy
    * manner.
    */
-  private synchronized void initializeLocalBlockWorkerClientPool() {
-    NetAddress localWorkerAddress =
-        getWorkerAddress(NetworkAddressUtils.getLocalHostName(ClientContext.getConf()));
-
-    // If the local worker is not available, do not initialize the local worker client pool.
-    if (localWorkerAddress == null) {
-      mLocalBlockWorkerClientPool = null;
-    } else {
-      mLocalBlockWorkerClientPool = new BlockWorkerClientPool(localWorkerAddress);
+  private void initializeLocalBlockWorkerClientPool() {
```

Comment:
question: can we put this method into static block? In that case, do we still need `mLocalBlockWorkerClientPoolInitialized` to guard whether this? And also simplify the logic of some methods below (they call this before doing stuff)?